src/app/pages/virtualization/components/all-instances/instance-list/instance-list.component.ts

```diff
@@ -0,0 +1,171 @@
+  protected readonly requireRoles = [Role.VirtInstanceWrite];
+  protected readonly searchQuery = signal<string>('');
+  protected readonly showMobileDetails = signal<boolean>(false);
+  protected readonly isMobileView = signal(false);
+  protected readonly window = inject<Window>(WINDOW);
+  protected readonly selection = new SelectionModel<string>(true, []);
+
+  protected readonly instances = this.store.instances;
+  protected readonly isLoading = this.store.isLoading;
+  protected readonly selectedInstance = this.store.selectedInstance;
+
+  protected readonly isAllSelected = computed(() => {
+    return this.selection.selected.length === this.instances().length;
+  });
+
+  protected readonly filteredInstances = computed(() => {
+    return this.instances()
+      .filter((instance) => {
+        return instance?.name?.toLocaleLowerCase()
+          .includes(this.searchQuery().toLocaleLowerCase());
+      });
+  });
+
+  protected readonly emptyConfig = signal<EmptyConfig>({
+    type: EmptyType.NoPageData,
+    title: this.translate.instant('No instances'),
+    message: this.translate.instant('Instances you created will automatically appear here.'),
+    large: true,
+  });
+
+  constructor(
+    private store: VirtualizationInstancesStore,
+    private router: Router,
+    private activatedRoute: ActivatedRoute,
+    private ws: WebSocketService,
+    private dialog: DialogService,
+    private translate: TranslateService,
+    private errorHandler: ErrorHandlerService,
+  ) {}
+
+  ngAfterViewInit(): void {
+    this.activatedRoute.paramMap.pipe(
+      untilDestroyed(this),
+    ).subscribe((params) => {
+      this.selectForDetails(params.get('id'));
+    });
+  }
+
+  onSearch(query: string): void {
+    this.searchQuery.set(query);
+  }
+
+  toggleAllChecked(checked: boolean): void {
+    if (checked) {
+      this.instances().forEach((instance) => this.selection.select(instance.id));
+    } else {
+      this.selection.clear();
+    }
+  }
+
+  viewDetails(instance: VirtualizationInstance): void {
+    this.selectForDetails(instance.id);
+
+    this.router.navigate(['/virtualization', 'view', instance.id]);
+
+    if (this.isMobileView()) {
+      this.showMobileDetails.set(true);
+
+      setTimeout(() => (this.window.document.getElementsByClassName('mobile-back-button')[0] as HTMLElement).focus(), 0);
+    }
+  }
+
+  private selectForDetails(instanceId: string): void {
+    if (!this.instances()?.length) {
+      return;
+    }
+
+    const selected = instanceId && this.instances().find((instance) => instance.id === instanceId);
+    if (selected) {
+      this.store.selectInstance(selected.id);
+      return;
+    }
+
+    this.selectFirstApp();
+  }
+
+  private selectFirstApp(): void {
+    const [first] = this.instances();
+
+    if (first) {
+      this.store.selectInstance(first.id);
+    }
+  }
+
+  closeMobileDetails(): void {
+    this.showMobileDetails.set(false);
+  }
+
+  restart(instanceId: string): void {
```

Comment:
Any particular reason to do it here and not in the child component?

Reply:
The idea is to separate logic from `ix-instance-row` using it solely for representation purposes.